File: day01/day01.py
nums = []
freq = {}
with open('input.txt', 'r') as f:
    lines = f.readlines()
    for line in lines:
        if len(line) == 0:
            break
        a, b = map(int, line.split())
        nums.append(a)
        if b not in freq:
            freq[b] = 0
        freq[b] += 1
sum = 0
for i in nums:
    if i in freq:
        sum += i * freq[i]
print(sum)

File: day01/day01.go
package main

import (
	"fmt"
	"os"
	"slices"
	"strconv"
)

func main() {
	f, err := os.Open("input.txt")
	defer f.Close()
	if err != nil {
		panic(err)
	}
	li := make([]int, 1000, 1000)
	ri := make([]int, 1000, 1000)
	sum := 0
	b := make([]byte, 14)
	// 5 digits + 3 spaces + 5 digits + \n
	for i := 0; i < 1000; i++ {
		_, err := f.Read(b)
		if err != nil {
			break
		}
		li[i], _ = strconv.Atoi(string(b[:5]))
		ri[i], _ = strconv.Atoi(string(b[8:13]))
	}
	slices.Sort(li)
	slices.Sort(ri)
	for i := 0; i < 1000; i++ {
		sum += abs(li[i] - ri[i])
	}
	fmt.Println(sum)
}

func abs(a int) int {
	if a < 0 {
		a = -a
	}
	return a
}
